pipenv/cli.py

```diff
@@ -2004,9 +2026,10 @@ def uninstall(
 @click.option('--python', default=False, nargs=1, help="Specify which version of Python virtualenv should use.")
 @click.option('--verbose', is_flag=True, default=False, help="Verbose mode.")
 @click.option('--requirements', '-r', is_flag=True, default=False, help="Generate output compatible with requirements.txt.")
+@click.option('--dev', '-d', is_flag=True, default=False, help="Generate output compatible with requirements.txt for the development dependencies.")
```

Comment:
@erinxocon I think with `--dev` it includes _also_ normal packages? Should change this text to "Include dev packages into requirements.txt output"? Now it makes one think that only dev packages will be outputted.

Reply:
Hmm, I think my assumption is wrong, as you wrote:

> pipenv lock -r -d will give you just the development deps and development vcs deps

Ok, then the help text is according to this specification.

Apparently it's impossible to create such requirements.txt files now, that would include _both_ standard and dev packages. I don't have use case for that, so I think it's ok. Someone else might want that, though.

Reply:
The current functionality separates them.  `--dev` does only development dependencies, and `-r` only does anything listed under the default section!  Is this not what was needed?

Reply:
If the all packages is needed I can add it at a later time!

Reply:
> The current functionality separates them. --dev does only development dependencies, and -r only does anything listed under the default section! Is this not what was needed?

Indeed, this change should fulfill my needs. Thanks :)

Reply:
@erinxocon Note that there is some inconsistence:

    pipenv install  # by default only standard packages, with --dev ALSO dev packages
    pipenv update  # by default only standard packages, with --dev ALSO dev packages
    pipenv lock -r  # by default only standard packages, with --dev ONLY dev packages

Reply:
In this case is that the wanted use case though? I'll let my fellow contributors put in their 2 cents!

Reply:
So to clarify, I needed `requirements.txt` file that excludes dev packages. After the change you've made here, I can do that with `pipenv lock -r` as you've instructed. So yeah, my use is now supported.

I have no opinion if `pipenv lock -r -d` should include _only_ dev packages or _also_ dev packages. I personally have no use case for neither. I was just pointing out that there's now slight inconsistency in API, compared to `install` and `update`. So you might want to consider changing `--dev` behavior here accordingly.

Reply:
I appreciate the input @tuukkamustonen!   A very keen eye on the api discrepancies!

Reply:
It seems `--dev` in `pipenv lock -r --dev` should be consistent with `pipenv install` and `pipenv update` indeed. In the future, if there's a need, additonal flag like `--only-dev` could be added.

Reply:
I just came here to post a bug stating that `lock -r -d` only outputs dev dependencies.  In isolation, I don't think there's anything inherently wrong with the current behavior, but the fact that `lock -d` is inconsistent with `install -d` and `update -d` makes it seem like a bug.  Not only that, but if I run `pipenv lock -h`, the help text for `--dev` says "Install both develop and default packages.", further adding to the confusion.